include/maths/common/CModel.h

```diff
@@ -329,6 +329,29 @@ struct MATHS_COMMON_EXPORT SModelProbabilityResult {
     SAnomalyScoreExplanation s_AnomalyScoreExplanation;
 };
 
+//! \brief The allocator interface.
+//!
+//! DESCRIPTION:\n
+//! The allocator interface is used to control the ability
+//! of a model to allocate new components. This is used to
+//! comply with the memory constraints of the system.
+class MATHS_COMMON_EXPORT CModelAllocator {
```

Comment:
This is very generic functionality. If we want to add allocators in the future it feels like we'll end up duplicating. Also this doesn't strictly allocate it manages the memory budget.

My inclination would be to factor this out into its own header. I would also rename to something like `CMemoryCircuitBreaker`.

Reply:
Done